packages/fiori/src/DynamicSideContent.js

```diff
@@ -0,0 +1,546 @@
+			contentHeight = this.breakpoint === this.sizeS && this.sideContentVisibility !== SideContentVisibility.AlwaysShow ? "100%" : "auto";
+
+		return {
+			dynamicSideContentRoot: {
+				"height": "100%",
+				"flex-wrap": this._mcSpan === "12" ? "wrap" : "nowrap",
+			},
+			mainContent: {
+				"height": mcSpan === this.span12 ? contentHeight : "100%",
+				"order": this.sideContentPosition === SideContentPosition.Begin ? 2 : 1,
+			},
+			sideContent: {
+				"height": scSpan === this.span12 ? contentHeight : "100%",
+				"order": this.sideContentPosition === SideContentPosition.Begin ? 1 : 2,
+			},
+		};
+	}
+
+	get accInfo() {
+		return {
+			"label": DynamicSideContent.i18nBundle.getText(DSC_SIDE_ARIA_LABEL),
+		};
+	}
+
+	get sizeS() {
+		return "S";
+	}
+
+	get sizeM() {
+		return "M";
+	}
+
+	get sizeL() {
+		return "L";
+	}
+
+	get sizeXL() {
+		return "XL";
+	}
+
+	get span0() {
+		return "0";
+	}
+
+	get span3() {
+		return "3";
+	}
+
+	get span4() {
+		return "4";
+	}
+
+	get span6() {
+		return "6";
+	}
+
+	get span8() {
+		return "8";
+	}
+
+	get span9() {
+		return "9";
+	}
+
+	get span12() {
+		return "12";
+	}
+
+	get spanFixed() {
+		return "fixed";
+	}
+
+	get containerWidth() {
+		return this.containerQuery ? this.getDomRef().clientWidth : window.innerWidth;
+	}
+
+	get breakpoint() {
+		const S_M_BREAKPOINT = 720,	// Breakpoint between S and M screen size
+			M_L_BREAKPOINT = 1024, // Breakpoint between M and L screen size
+			L_XL_BREAKPOINT = 1440; // Breakpoint between L and XL screen size
+		let size;
+
+		if (this.containerWidth <= S_M_BREAKPOINT) {
+			size = this.sizeS;
+		} else if (this.containerWidth > S_M_BREAKPOINT && this.containerWidth <= M_L_BREAKPOINT) {
+			size = this.sizeM;
+		} else if (this.containerWidth > M_L_BREAKPOINT && this.containerWidth <= L_XL_BREAKPOINT) {
+			size = this.sizeL;
+		} else {
+			size = this.sizeXL;
+		}
+
+		return size;
+	}
+
+	toggleContents() {
+		if (this.breakpoint === this.sizeS && this.sideContentVisibility !== SideContentVisibility.AlwaysShow) {
+			this._toggled = !this._toggled;
+		}
+	}
+
+	handleResize() {
+		this._resizeContents();
+	}
+
+	onAfterRendering() {
+		this._resizeContents();
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResizeBound);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResizeBound);
+	}
+
+	_resizeContents() {
+		const contentWidth = Math.ceil((33.333 / 100) * this.containerWidth); // for SideContentFallDown.OnMinimumWidth case
+		let mainSize,
+			sideSize,
+			sideVisible;
+
+		// initial set contents sizes
+		switch (this.breakpoint) {
+		case this.sizeS:
+			mainSize = this.span12;
+			sideSize = this.span12;
+			break;
+		case this.sizeM:
+			if (this.sideContentFallDown === SideContentFallDown.BelowXL
+				|| this.sideContentFallDown === SideContentFallDown.BelowL
+				|| (contentWidth <= 320 && this.sideContentFallDown === SideContentFallDown.OnMinimumWidth)) {
```

Comment:
magic number (320) - please create a constant for this too (and group it with the breakpoints constants at the top of the file - see the comment about them)

Reply:
Done